### PR TITLE
fix: team billing on batch operations

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

- Fixes CAL-3867

A quick and dirty fix preceding a full refactor on #15366 

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [ ] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

## How should this be tested?

- Having Stripe keys setup, batch invite and batch remove members from a team/org. Stripe subscription should match the correct membership count.

### CODE REVIEW

File: packages/trpc/server/routers/viewer/organizations/bulkDeleteUsers.handler.ts [hunk 1]
```diff
@@ -62,7 +63,7 @@ export async function bulkDeleteUsersHandler({ ctx, input }: BulkDeleteUsersHand
   // We do this in a transaction to make sure that all memberships are removed before we remove the organization relation from the user
   // We also do this to make sure that if one of the queries fail, the whole transaction fails
   await prisma.$transaction([removeProfiles, deleteMany, removeOrgrelation]);
-
+  await updateQuantitySubscriptionFromStripe(currentUser.organizationId);
```
Comment (zomars) [inline packages/trpc/server/routers/viewer/organizations/bulkDeleteUsers.handler.ts:66]: Batch deleting didn't update Stripe subscription before.

File: packages/trpc/server/routers/viewer/teams/inviteMember/inviteMember.handler.ts [hunk 2]
```diff
@@ -151,7 +151,7 @@ export const inviteMemberHandler = async ({ ctx, input }: InviteMemberOptions) =
         isOrg: input.isOrg,
       });
     });
-    sendEmails(sendVerifEmailsPromises);
+    await sendEmails(sendVerifEmailsPromises);
```
Comment (zomars) [inline packages/trpc/server/routers/viewer/teams/inviteMember/inviteMember.handler.ts:154]: I think this was causing a race condition that caused to report a wrong membership count

File: packages/trpc/server/routers/viewer/teams/inviteMember/inviteMember.handler.ts [hunk 3]
```diff
@@ -166,20 +166,20 @@ export const inviteMemberHandler = async ({ ctx, input }: InviteMemberOptions) =
...
   });
 
-  if (IS_TEAM_BILLING_ENABLED) {
-    if (team.parentId) {
-      await updateQuantitySubscriptionFromStripe(team.parentId);
-    } else {
-      await updateQuantitySubscriptionFromStripe(input.teamId);
-    }
-  }
+  await handleSubscriptionUpdates(team.parentId || input.teamId);
```
Comment (zomars) [inline packages/trpc/server/routers/viewer/teams/inviteMember/inviteMember.handler.ts:169]: Refactored for denesting

**Review by emrysal — APPROVED** [review]
Looks good - indeed quick & dirty but straightforward.

**Pull request merged**